pkg/syncer/metadatasyncer.go

```diff
@@ -517,33 +546,46 @@ func updatePodMetadata(ctx context.Context, pod *v1.Pod, metadataSyncer *metadat
 func csiPVCUpdated(ctx context.Context, pvc *v1.PersistentVolumeClaim, pv *v1.PersistentVolume, metadataSyncer *metadataSyncInformer) {
 	log := logger.GetLogger(ctx)
 	volumeFound := false
-	// Following wait poll is required to avoid race condition between pvcUpdated and pvUpdated
-	// This helps avoid race condition between pvUpdated and pvcUpdated handlers when static PV and PVC is created almost
-	// at the same time using single YAML file.
-	err := wait.Poll(5*time.Second, time.Minute, func() (bool, error) {
-		queryFilter := cnstypes.CnsQueryFilter{
-			VolumeIds: []cnstypes.CnsVolumeId{{Id: pv.Spec.CSI.VolumeHandle}},
+	var volumeHandle string
+	var err error
+	if metadataSyncer.configInfo.Cfg.FeatureStates.CSIMigration && pv.Spec.VsphereVolume != nil {
+		volumeHandle, err = GetVolumeID(ctx, pv)
```

Comment:
When `migrated` annotation is applied on PVC and PV, they will race for registering volume. We should avoid this race.

Can we maintain an internal queue and dispatch registration in credence. We may have 10k volumes, in that case, we will see strom of volume registration calls.

Reply:
if informer is dispatching updates one by one, then ignore my comment regarding maintaining the internal queue.

also regarding the race between PVC and PV to register the same volume, VC will take care of this. I thinking about avoiding multiple calls to the vCenter for same volume when migration annotation is getting added. We can address this later

Reply:
Yes. The informer dispatches the updates one by one.

Regarding multiple updates to same volume to CNS: We are handling this by checking the faultType == alreadyRegisteredFault in volumeManager. We need not do more processing to avoid multiple calls